docs/token-contracts/fa2/2-fa2-nft-tutorial.md

```diff
@@ -0,0 +1,533 @@
+
+originating new NFT contract...
+originated NFT collection KT1XP3RE6S9t44fKR9Uo5rAfqHvHXu9Cy7fh
+```
+
+### Inspecting The NFT Contract
+
+Using `KT1..` address of the NFT contract created by the `mint` command, we can
+inspect token metadata and balances (i. e. which addresses own the tokens).
+
+#### Inspect Token Metadata
+
+`show-meta` command requires the following parameters:
+
+- `--nft` address of the FA2 NFT contract to inspect
+- `--signer` alias on behalf of which contract is inspected
+- `--tokens` a list of token IDs to inspect
+
+```sh
+$ tznft show-meta --nft <nft_address> --signer <alias> --tokens <token_id_list>
+```
+
+Example:
+
+```sh
+$ tznft show-meta --nft KT1XP3RE6S9t44fKR9Uo5rAfqHvHXu9Cy7fh --signer bob --tokens 0 1
+
+token_id: 0	symbol: T1	name: My Token One	extras: { }
+token_id: 1	symbol: T2	name: My Token Two	extras: { }
+```
+
+#### Inspect Token Balances
+
+`show-balance` command requires the following parameters:
+
+- `--nft` address of the FA2 NFT contract to inspect
+- `--signer` alias on behalf of which contract is inspected
+- `--owner` alias of the token owner to check balances
+- `--tokens` a list of token IDs to inspect
+
+```sh
+$ tznft show-balance --nft <nft_address> --signer <alias> --owner <alias> --tokens <token_id_list>
+```
+
+Example 1, check `bob`'s balances:
+
+```sh
+$ tznft show-balance --nft KT1XP3RE6S9t44fKR9Uo5rAfqHvHXu9Cy7fh --signer bob --owner bob --tokens 0 1
+
+querying NFT contract KT1XP3RE6S9t44fKR9Uo5rAfqHvHXu9Cy7fh using balance inspector KT1Pezr7JjgmrPcPhpkbkH1ytG7saMZ34sfd
+requested NFT balances:
+owner: tz1YPSCGWXwBdTncK2aCctSZAXWvGsGwVJqU	token: 0	balance: 1
+owner: tz1YPSCGWXwBdTncK2aCctSZAXWvGsGwVJqU	token: 1	balance: 1
+```
+
+Example 2, check `alice` balances:
+
+```sh
+$ tznft show-balance --nft KT1XP3RE6S9t44fKR9Uo5rAfqHvHXu9Cy7fh --signer bob --owner alice --tokens 0 1
+
+querying NFT contract KT1XP3RE6S9t44fKR9Uo5rAfqHvHXu9Cy7fh using balance inspector KT1Pezr7JjgmrPcPhpkbkH1ytG7saMZ34sfd
+requested NFT balances:
+owner: tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb	token: 0	balance: 0
+owner: tz1VSUr8wwNhLAzempoch5d6hLRiTh8Cjcjb	token: 1	balance: 0
+```
+
+### Tokens With External Metadata
+
+Token metadata can store a reference to some external document and/or image.
+This tutorial supports storing external data on [IPFS](https://ipfs.io) and keeping
+an IPFS hash as a part of the token metadata.
+
+Let's create a single NFT token which references an image on IPFS.
+
+1. Upload your image to IPFS and obtain an image file hash. There are
+   multiple ways to do that. One of the possible solutions is to install the
+   [IPFS Companion](https://github.com/ipfs-shipyard/ipfs-companion) web plugin and
+   upload an image file from there. You can upload multiple images and/or documents
+   if you plan to create a collection of multiple NFTs.
+
+2. Copy the IPFS file hash code (`CID`). For this example we will use
+   `QmRyTc9KbD7ZSkmEf4e7fk6A44RPciW5pM4iyqRGrhbyvj`
```

Comment:
This could be a link: [https://ipfs.io/ipfs/QmRyTc9KbD7ZSkmEf4e7fk6A44RPciW5pM4iyqRGrhbyvj](https://ipfs.io/ipfs/QmRyTc9KbD7ZSkmEf4e7fk6A44RPciW5pM4iyqRGrhbyvj)